sql-cli/sql_cli/__main__.py

```diff
@@ -37,111 +41,105 @@ def about() -> None:
 
 @app.command(help="Generate the Airflow DAG from a directory of SQL files.")
 def generate(
-    directory: Path = typer.Argument(
+    workflow_name: str = typer.Argument(
         default=...,
         show_default=False,
-        exists=True,
-        help="directory containing the sql files.",
+        help="name of the workflow directory within workflows directory.",
     ),
-    dags_directory: Path = typer.Argument(
-        default=...,
-        show_default=False,
-        exists=True,
-        help="directory for the generated DAGs.",
+    env: str = typer.Option(
+        default="default",
+        help="environment to run in",
+    ),
+    project_dir: Path = typer.Option(
+        None, dir_okay=True, metavar="PATH", help="(Optional) Default: current directory.", show_default=False
     ),
 ) -> None:
-    dag_file = generate_dag(directory, dags_directory)
+    project_dir_absolute = project_dir.resolve() if project_dir else Path.cwd()
+    project = Project(project_dir_absolute)
+    project.load_config(env)
+
+    # Since we are using the Airflow ORM to interact with connections, we need to tell Airflow to use our airflow.db
+    # The usual route is to set $AIRFLOW_HOME before Airflow is imported. However, in the context of the SQL CLI, we
+    # decide this during runtime, depending on the project path and SQL CLI configuration.
+    airflow_meta_conn = retrieve_airflow_database_conn_from_config(project.directory / project.airflow_home)
+    set_airflow_database_conn(airflow_meta_conn)
+
+    dag_file = generate_dag(
+        directory=project.directory / project.workflows_directory / workflow_name,
+        dags_directory=project.airflow_dags_folder,
+    )
     rprint("The DAG file", dag_file.resolve(), "has been successfully generated. 🎉")
 
 
 @app.command(
-    help="Validate Airflow connection(s) provided in the configuration file for the given environment."
+    help="""
+    Validate Airflow connection(s) provided in the configuration file for the given environment.
+    """
 )
 def validate(
-    environment: str = typer.Argument(
+    project_dir: Path = typer.Argument(
+        None, dir_okay=True, metavar="PATH", help="(Optional) Default: current directory.", show_default=False
+    ),
+    env: str = typer.Option(
         default="default",
-        help="environment to validate",
+        help="(Optional) Environment used to declare the connections to be validated",
     ),
-    connection: str = typer.Argument(
+    connection: str = typer.Option(
         default=None,
-        help="airflow connection id to validate",
+        help="(Optional) Identifier of the connection to be validated. By default checks all the env connections.",
     ),
 ) -> None:
-    validate_connections(environment=environment, connection_id=connection)
```

Comment:
At the moment, we have a bug.

Steps to reproduce (to be logged as a separate ticket):
```
$ poetry run flow init final
Initialized an Astro SQL project at 

$ poetry run flow validate final

Validating connection(s) for environment 'default'
Validating connection sqlite_conn               
PASSED

# add the following lines to final/config/default/configuration.yml 
- conn_id: sqlite_conn_invalid
  conn_type: sqlite
  host: /home/tati/Code/astro-sdk/sql-cli/final/data/imdb2.db
  login: null
  password: null
  schema: null

$ poetry run flow validate final

Validating connection(s) for environment 'default'
Validating connection sqlite_conn               
PASSED
Validating connection sqlite_conn_invalid       
PASSED
```
We should expect the validation for `sqlite_conn_invalid` to fail. We should solve this as part of 0.1